pkg/network/podinterfacenamescheme/podinterfacenamescheme.go

```diff
@@ -19,4 +19,37 @@
 
 package podinterfacenamescheme
 
+import (
+	"fmt"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/multus"
+)
+
 const PrimaryPodInterfaceName = "eth0"
+
+type PodInterfaceNamer map[string]string
```

Comment:
Why is it necessary to define new type?

Reply:
not necessary, I think this is the best approach. 
why not do a new type?

Reply:
Can you define receivers to the map directly?
BTW, "namer" sounds like an interface name.

Reply:
I don't think it's possible to use a receiver on a non-named type.

>BTW, "namer" sounds like an interface name

not sure what you mean.. :)

Reply:
https://go.dev/doc/effective_go#interface-names

Reply:
> not necessary, I think this is the best approach. why not do a new type?

@RamLavi if we remove the type definition there is no difference in the way it will be used, I think it doesn't improve reliability.

I understand that your intention was to make the code self-explanatory by adding methods for it that suggest
what will the naming scheme include (`mapDefaultNetworkToPodInterfaceName `mapMultusNonDefaultNetworksToPodInterfaceName`)
Note that if you get rid of the type, you can still have those methods, in just that they won't do the assignment to the map..